### ISSUE TITLE
Warn the user if the GitRepository reference will be empty

### ISSUE THREAD

**Issue by palemtnrider** [weaveworks/pctl#295]
**What were you trying to accomplish?**

Run a pctl add to an environment bootstraped by Weave GitOps. Flux is installed in the wego-system namespace

**What happened?**

The kustomize-flux.yaml referred to a GitRepository with a blank name and no namespace.
```yaml
  sourceRef:
    kind: GitRepository
    name: ""
```
**How to reproduce it?**

don't provide `--git-repository` or a bootstrap config file when running `pctl add`

**Logs**

**Anything else we need to know?**

This isn't major as the workaround is easy enough.  However, if we can warn the user that the profile won't be pulled into the cluster that can save some debugging time.

**Versions**
0.7.0

**Comment by aclevername** [comment]
Thanks for opening the issue @palemtnrider ! Huh this is definitely a bug, we used to always error before if `--git-repository` wasn't provided. This logic probably got broken with the introduction of the pctl config file. I will reproduce and fix :smile:

**Comment by aclevername** [comment]
Huh I couldn't reproduce this @palemtnrider. With no bootstrap file locally:
```
./pctl add --name jake-rocks nginx-catalog/weaveworks-nginx/v0.1.1
► generating profile installation from source: catalog entry nginx-catalog/weaveworks-nginx/v0.1.1
✗ failed to make artifacts: in case of local resources, the flux gitrepository object's details must be provided
```

Could you provide the command you ran to reproduce this problem?

**Comment by aclevername** [comment]
Actually I managed to reproduce this after thinking through the problem some more. We used to have `--git-repository` be optional in the small scenario where a profile consists only of a remote helm chart, e.g. https://github.com/weaveworks/nginx-profile. But when we changed to have it mandatory in order to do `dependsOn` this logic never got adjusted, causing the bug you found! 

I will open a PR to fix :smile:
